### PR TITLE
docs: Update naming conventions and added examples

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Added examples and updated naming conventions in docs.

## Motivation and Context

Updated docs for better naming conventions.

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

**Review by bernard-eugine — APPROVED** [review]

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -623,6 +623,7 @@ pub enum SupportedWallets {
     Gpay,
 }
 
+/// The hyperswitch code of all the supported banks
```
Comment (SanchithHegde) [inline crates/api_models/src/enums.rs:626]:
```suggestion
/// Names of banks supported by HyperSwitch
```

File: crates/api_models/src/payment_methods.rs [hunk 2]
```diff
@@ -162,27 +162,52 @@ pub struct CardDetailFromLocker {
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema, PartialEq, Eq)]
 pub struct PaymentExperienceTypes {
+    /// The payment experience enabled
+    #[schema(value_type = Option<PaymentExperience>, example = "redirect_to_url")]
     pub payment_experience_type: api_enums::PaymentExperience,
+
+    /// The list of eligible connectors for a given payment experience
+    #[schema(value_type = Vec<String>, example = json!(["stripe", "adyen"]))]
```
Comment (SanchithHegde) [inline crates/api_models/src/payment_methods.rs:170]: `value_type` need not be specified here.

File: crates/api_models/src/payment_methods.rs [hunk 3]
```diff
@@ -162,27 +162,52 @@ pub struct CardDetailFromLocker {
...
 }
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema, PartialEq, Eq)]
 pub struct CardNetworkTypes {
+    /// The card network enabled
+    #[schema(value_type = Option<CardNetwork>, example = "Visa")]
     pub card_network: api_enums::CardNetwork,
+
+    /// The list of eligible connectors for a given card network
+    #[schema(value_type = Vec<String>, example = json!(["stripe", "adyen"]))]
```
Comment (SanchithHegde) [inline crates/api_models/src/payment_methods.rs:181]: Same here.

File: crates/api_models/src/payment_methods.rs [hunk 4]
```diff
@@ -162,27 +162,52 @@ pub struct CardDetailFromLocker {
...
-#[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, ToSchema)]
 pub struct ResponsePaymentMethodsEnabled {
+    /// The payment method enabled
+    #[schema(value_type = PaymentMethod)]
     pub payment_method: api_enums::PaymentMethod,
+
+    /// The list of payment method types enabled for a connector account
+    #[schema(value_type = Vec<ResponsePaymentMethodTypes>)]
     pub payment_method_types: Vec<ResponsePaymentMethodTypes>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payment_methods.rs:211]: Same here, `value_type` need not be specified as all the types are either directly imported or are in the same module.

File: crates/api_models/src/payment_methods.rs [hunk 5]
```diff
@@ -402,22 +427,6 @@ pub struct ListPaymentMethodResponse {
...
-//             payment_experience: pm_type.payment_experience,
-//             connector,
-//             card_networks: pm_type.card_networks,
-//             payment_method,
-//         }
-//     }
-// }
-
 #[derive(Eq, PartialEq, Hash, Debug, serde::Deserialize, ToSchema)]
 pub struct ListPaymentMethod {
```
Comment (SanchithHegde) [inline crates/api_models/src/payment_methods.rs:431]: Consider renaming this and other types in this module to match the naming conventions.

File: crates/api_models/src/payments.rs [hunk 6]
```diff
@@ -369,19 +369,28 @@ pub struct Card {
...
     /// The card holder's name
     #[schema(value_type = String, example = "John Test")]
     pub card_holder_name: Secret<String>,
+
     /// The CVC number for the card
     #[schema(value_type = String, example = "242")]
     pub card_cvc: Secret<String>,
+
+    /// The name of the issuer of card
+    #[schema(value_type = Option<String>, example = "chase")]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:390]: `value_type` need not be specified here.

File: crates/api_models/src/payments.rs [hunk 7]
```diff
@@ -468,49 +477,57 @@ impl From<&PaymentMethodData> for AdditionalPaymentData {
 pub enum BankRedirectData {
     Eps {
         /// The billing details for bank redirection
+        #[schema(value_type = BankRedirectBilling)]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:480]: Same here.

File: crates/api_models/src/payments.rs [hunk 8]
```diff
@@ -468,49 +477,57 @@ impl From<&PaymentMethodData> for AdditionalPaymentData {
...
         billing_details: BankRedirectBilling,
-        #[schema(value_type = BankNames)]
+
+        /// The hyperswitch bank code for eps
+        #[schema(value_type = BankNames, example = "triodos_bank")]
         bank_name: api_enums::BankNames,
     },
     Giropay {
         /// The billing details for bank redirection
+        #[schema(value_type = BankRedirectBilling)]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:489]: Same here.

File: crates/api_models/src/payments.rs [hunk 9]
```diff
@@ -468,49 +477,57 @@ impl From<&PaymentMethodData> for AdditionalPaymentData {
...
         billing_details: BankRedirectBilling,
     },
     Ideal {
         /// The billing details for bank redirection
+        #[schema(value_type = BankRedirectBilling)]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:494]: Same here.

File: crates/api_models/src/payments.rs [hunk 10]
```diff
@@ -888,30 +942,47 @@ pub struct PaymentsResponse {
...
     /// Time less than or equals to the payment created time
+    #[schema(example = "2022-09-10T10:11:12Z")]
     #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
     #[serde(rename = "created.lte")]
     pub created_lte: Option<PrimitiveDateTime>,
+
     /// Time greater than or equals to the payment created time
+    #[schema(example = "2022-09-10T10:11:12Z")]
     #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
     #[serde(rename = "created.gte")]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:987]: Unrelated, please merge the `serde` attributes here.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/api_models/src/payment_methods.rs [hunk 11]
```diff
@@ -402,24 +424,8 @@ pub struct ListPaymentMethodResponse {
     pub payment_methods: Vec<ResponsePaymentMethodsEnabled>,
 }
 
-// impl ResponsePaymentMethodTypes {
```
Comment (jarnura) [inline crates/api_models/src/payment_methods.rs:405]: Remove commented codes

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/api_models/src/payment_methods.rs (hunk 11 above)
Comment (dracarys18) [reply crates/api_models/src/payment_methods.rs:405]: It's not there in the code

Comment (jarnura) [reply crates/api_models/src/payment_methods.rs:405]: My bad

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by bernard-eugine — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**